hummingbot/hummingbot
# feat/bitget connector

**Before submitting this PR, please make sure**:


**A description of the changes proposed in the pull request**:
Implementation of the perpetual connector to Bitget exchange.
The connector supports the three available product types:
- UMCBL
- CMCBL
- DMCBL

For UMCBL the collateral token is always USDT (the quote token for the trading pairs)
For CMCBL the collateral token is always USDC
For DMCBL the collateral token is always the base asset of the trading pair

The API documentation is https://bitgetlimited.github.io/apidoc/en/spot/#welcome

**Tests performed by the developer**:
Added unit tests for all connector components. All tests passing in green.

- Build Docker image and create instance using it
- Start the client without any errors
- Connect invalid/valid API keys
- Pull balance without any issue
- Create/start strategy without any errors
- No issue on creation and cancellation of orders
- Open position is recognized upon restart
- status/orderbook is sync
- Filled trades + fee are logged correctly
- Checked funding payment on time
- Run the bot on multiple order levels
- Checking invalid API key gets rejected with a correct error message to end user
- Connecting correct and a valid API key
- Checking balance thereafter if it is getting displayed or not
- Cross verifying it with an exchange if the balance is pulled correctly or not
- Created a Perpetual market-making strategy with USDC quote tokens. The strategy runs as expected.
- Created a Perpetual market-making strategy with USD quote tokens. 



**Tips for QA testing**:
Please validate the new connector with any strategy using perpetual connectors.
Include all validations for new connectors.



HIP Link: https://snapshot.org/#/hbot-ip.eth/proposal/0x3644b2a83d69310201f24d44060c2edfcaf7c0cfab511f6f9bf211faa6c8f894
